[Mono.Android-Tests] Remove obsolete `.shproj` file.

jpobst opened the pull request:
`Mono.Android-Tests` still uses a `.shproj` file from the time when we had both a Classic XA and .NET version.  However, VS does not seem to support this anymore, and none of the included files appear in the IDE, making it hard to write new tests.

Remove the `.shproj` file and update the `.csproj` file to directly point to needed files.

Also do some other cleanups like removing tests that are no longer run and fixing directory structure.

Verified that both the `main` version and this PR run 885 tests on CI.

jonathanpeppers reviewed (commented):
Did I misread the diff, or did a couple tests get deleted on accident?

jpobst commented:
I removed tests that are not being run today in `main`, assuming they were obsolete and removed on purpose.  I can try enabling them and see if they pass or not.

jonathanpeppers commented:
The two above look like they should pass, but `Mono.Data.Sqlite/SqliteTests.cs` can be deleted.

jpobst commented:
> The two above look like they should pass, but Mono.Data.Sqlite/SqliteTests.cs can be deleted.

👍 Tests enabled.
